frame/support/src/storage/bounded_vec.rs

```diff
@@ -39,60 +44,60 @@ use sp_std::{marker::PhantomData, prelude::*};
 /// this bound is respected.
 #[derive(Encode, scale_info::TypeInfo)]
 #[scale_info(skip_type_params(S))]
-pub struct BoundedVec<T, S>(Vec<T>, PhantomData<S>);
+pub struct BoundedVec<T, B, S: Get<B>>(Vec<T>, PhantomData<B>, PhantomData<S>);
 
 /// A bounded slice.
 ///
 /// Similar to a `BoundedVec`, but not owned and cannot be decoded.
 #[derive(Encode, scale_info::TypeInfo)]
 #[scale_info(skip_type_params(S))]
-pub struct BoundedSlice<'a, T, S>(&'a [T], PhantomData<S>);
+pub struct BoundedSlice<'a, T, B, S: Get<B>>(&'a [T], PhantomData<B>, PhantomData<S>);
```

Comment:
similarly

Reply:
There's no need to have two phantom datas, you can wrap multiple types in one with a tuple, like `PhantomData<(A, B, C)>`

Reply:
I understand now. This allows to settle the other suggestions as well. Thank you !